src/catalog/sequence_catalog.cpp

```diff
@@ -0,0 +1,299 @@
+  auto val3 = type::ValueFactory::GetBigIntValue(seq_increment);
+  auto val4 = type::ValueFactory::GetBigIntValue(seq_max);
+  auto val5 = type::ValueFactory::GetBigIntValue(seq_min);
+  auto val6 = type::ValueFactory::GetBigIntValue(seq_start);
+  auto val7 = type::ValueFactory::GetBooleanValue(seq_cycle);
+  // When insert value, seqval = seq_start
+  auto val8 = type::ValueFactory::GetBigIntValue(seq_start);
+
+  tuple->SetValue(ColumnId::SEQUENCE_OID, val0, pool);
+  tuple->SetValue(ColumnId::DATABSE_OID, val1, pool);
+  tuple->SetValue(ColumnId::SEQUENCE_NAME, val2, pool);
+  tuple->SetValue(ColumnId::SEQUENCE_INC, val3, pool);
+  tuple->SetValue(ColumnId::SEQUENCE_MAX, val4, pool);
+  tuple->SetValue(ColumnId::SEQUENCE_MIN, val5, pool);
+  tuple->SetValue(ColumnId::SEQUENCE_START, val6, pool);
+  tuple->SetValue(ColumnId::SEQUENCE_CYCLE, val7, pool);
+  tuple->SetValue(ColumnId::SEQUENCE_VALUE, val8, pool);
+
+  // Insert the tuple
+  return InsertTuple(std::move(tuple), txn);
+}
+
+/* @brief   Delete the sequence by name.
+ * @param   database_oid  the databse_oid associated with the sequence
+ * @param   sequence_name the name of the sequence
+ * @param   txn           current transaction
+ * @return  ResultType::SUCCESS if the sequence exists, throw exception
+ * otherwise.
+ */
+ResultType SequenceCatalog::DropSequence(const std::string &database_name,
+                                         const std::string &sequence_name,
+                                         concurrency::TransactionContext *txn) {
+  if (txn == nullptr) {
+    throw CatalogException("Transaction is invalid!");
+  }
+
+  auto database_object =
+      Catalog::GetInstance()->GetDatabaseObject(database_name, txn);
+
+  oid_t sequence_oid = Catalog::GetInstance()
+      ->GetSystemCatalogs(database_object->GetDatabaseOid())
+      ->GetSequenceCatalog()
+      ->GetSequenceOid(sequence_name, database_object->GetDatabaseOid(), txn);
+  if (sequence_oid == INVALID_OID) {
+    throw SequenceException(
+        StringUtil::Format("Sequence %s does not exist!",
+                           sequence_name.c_str()));
+  }
+
+  LOG_INFO("sequence %d will be deleted!", sequence_oid);
+
+  oid_t database_oid = database_object->GetDatabaseOid();
+  DeleteSequenceByName(sequence_name, database_oid, txn);
+
+  return ResultType::SUCCESS;
+}
+
+/* @brief   Delete the sequence by name. The sequence is guaranteed to exist.
+ * @param   database_oid  the databse_oid associated with the sequence
+ * @param   sequence_name the name of the sequence
+ * @param   txn           current transaction
+ * @return  The result of DeleteWithIndexScan.
+ */
+bool SequenceCatalog::DeleteSequenceByName(
+    const std::string &sequence_name, oid_t database_oid,
+    concurrency::TransactionContext *txn) {
+  oid_t index_offset = IndexId::DBOID_SEQNAME_KEY;
+  std::vector<type::Value> values;
+  values.push_back(type::ValueFactory::GetIntegerValue(database_oid).Copy());
```

Comment:
Why do we need Copy() here?

Reply:
Just following conventions from other files.